pkg/state/chart_dependency.go

```diff
@@ -260,7 +269,7 @@ func getUnresolvedDependenciess(st *HelmState) (string, *UnresolvedDependencies,
 			url = fmt.Sprintf("oci://%s", url)
 		}
 
-		if err := unresolved.Add(chart, url, r.Version); err != nil {
+		if err := unresolved.Add(chart, url, r.Version, chartDependenciesAlias(r.Namespace, r.Name)); err != nil {
```

Comment:
This started looking a bit odd to me- the idea behind `unresolved.Add` is that you add the dependency to the dependencies list only if it is not contained in the list yet.

So, usually, the first argument, `chart,` is a unique ID to look for duplicates. With this fix, the unique ID can be `alias` if exists. `Alias` exists only when two or more charts share the same `chart` (but different versions).

So probably this should be

```suggestion
		if err := unresolved.Add(chart, url, r.Version); err != nil {
```

with `Add` magically selecting which ID to use, `chart` or `alias`, OR

```suggestion
		if err := unresolved.Add(id, url, r.Version); err != nil {
```

where id is `chart` if there is only one release that depends on that chart, or `alias` if there are two or more releases share the `chart`.

Reply:
The problem of not doing that would, we end up unable to reuse already resolved dependency, because virtually every release has its own unique chart of `<namespace>-<release>`

Reply:
```
func (d *UnresolvedDependencies) Add(chart, url, versionConstraint, alias string) error {
	dep := unresolvedChartDependency{
		ChartName:         chart,
		Repository:        url,
		VersionConstraint: versionConstraint,
		Alias:             alias,
	}
	if !d.contains(dep) {
		return d.add(dep)
	}
	return nil
}
```
the `Add` func needs `chart` args. so we can remove that. 

@mumoshu

Reply:
Hey 👋 
We need both ChartName and Alias. ChartName is needed to generate a Helm Chart.lock file used when `helmfile deps`